src/lib/components/composites/project-components/CreateProjectDialog.svelte

```diff
@@ -0,0 +1,273 @@
+    import { goto } from "$app/navigation";
+    import CircleAlert from "lucide-svelte/icons/circle-alert";
+    import { getNames } from "$lib/utils/common-helper";
+    import type { ValidationResult } from "$lib/model/general";
+
+    // at the beginning the dialog should not be open
+    let open: boolean = $state(false);
+
+    let isServerStillCreatingProject = $state(false);
+    let projectWasCreated = $state(false);
+    let projectId = $state<number | undefined>(undefined);
+
+    let isErrorOnProjectCreation = $state(false);
+    let isErrorOnUsersLoading = $state(false);
+
+    let projectNameInput: Input;
+    let membersInput: string[] = $state([]);
+
+    // TODO: check, whether this solution scales as the backend contains hundreds / thousands of users
+    // list of possible members that can be invited
+    let initialPossibleMembers: User[] = $state([]);
+    let possibleMembers: User[] = $derived(
+        initialPossibleMembers.filter((member) => !membersInput.includes(member.email)),
+    );
+
+    onMount(async () => {
+        try {
+            // TODO: exchange by call to store or so, if login etc. is completely implemented
+            let thisUser: User = await BackendController.getInstance().thisUser().get();
+            initialPossibleMembers = await BackendController.getInstance()
+                .getUsers()
+                .then((users) => users.filter((user) => user.id !== thisUser.id));
+        } catch (error) {
+            isErrorOnUsersLoading = true;
+            console.error(`Could not get users from server (${error})`);
+        }
+    });
+
+    /**
+     * Filters all possible members by checking, whether their name or email contains the search string.
+     *
+     * Furthermore, the filtered list is sorted by the score from the FZF algorithm, i.e.
+     * the members with the best matching name or email are at the beginning of the list (and will
+     * appear at the top of the suggestions list).
+     *
+     * @param input the content of the input field, i.e. the search string
+     * @returns list of "name \<email\>" (sorted) representations of users that can be invited
+     */
+    function filterPossibleMembers(input: string): string[] {
+        const fzf = new Fzf(possibleMembers.map((user) => `${getNames([user])} <${user.email}>`));
+        return fzf.find(input.toLowerCase()).map((result) => result.item);
+    }
+
+    /**
+     * Checks, whether a given input is a valid name of a registered user or an email.
+     */
+    function validateInput(input: string): ValidationResult {
+        if (!Schema.email.safeParse(input.trim()).success) {
+            if (!possibleMembers.map((member) => getNames([member])).includes(input.trim())) {
+                return { success: false, error: "Please enter a valid name or email." };
+            }
+            if (
+                possibleMembers.filter((member) => getNames([member]) === input.trim()).length > 1
+            ) {
```

Comment:
Here, we basically run checks on the same result, so we can store the result before evaluating it.
```suggestion
            const matchingMembers = possibleMembers.filter(
                (member) => getNames([member]) === input.trim(),
            );
            if (matchingMembers.length === 0) {
                return { success: false, error: "Please enter a valid name or email." };
            }
            if (matchingMembers.length > 1) {
```

Reply:
Good idea, I will integrate your suggestion.